src/core/gui/XournalView.cpp

```diff
@@ -287,6 +289,23 @@ auto XournalView::onKeyPressEvent(GdkEventKey* event) -> bool {
         return true;
     }
 
+    // Switch color on number key
+    auto colors = control->getWindow()->getToolMenuHandler()->getColorToolItems();
+    if (!colors) {
+        return false;
+    }
+    for (size_t i = 0; i < colors->size(); i++) {
```

Comment:
The upper bound should be set to `std::min(colors->size(), 10)` since there are at most 10 number keys.

Also, it is possible to avoid the loop by checking whether `GDK_KEY_0 <= event->keyval < GDK_KEY_0 + std::min(10, colors->size())` and then computing the index into `colors` using `std::min(colors.size() - 1, (9 + (event->keyval - GDK_KEY_0)) % 10)`.  Note that the bounds check will force `colors.size() > 0`, so this should be safe even if there are no colors in the toolbar.